libpromises/evalfunction.c

```diff
@@ -2917,9 +2928,10 @@ static FnCallResult FnCallMergeData(EvalContext *ctx, ARG_UNUSED const Policy *p
 
     for (const Rlist *arg = args; arg; arg = arg->next)
     {
-        if (args->val.type != RVAL_TYPE_SCALAR)
+        if (args->val.type != RVAL_TYPE_SCALAR &&
+            args->val.type != RVAL_TYPE_CONTAINER)
         {
-            Log(LOG_LEVEL_ERR, "%s: argument '%s' is not a variable reference", fp->name, RlistScalarValue(arg));
+            Log(LOG_LEVEL_ERR, "%s: argument '%s' is not a variable reference", fp->name, RlistScalarValueSafe(arg));
```

Comment:
I think the `RlistScalarValueSafe()` here is guaranteed to be `[not printable]`, due to the check above. Might as well be removed.